src/AutoRest.CSharp/Generation/Writers/LowLevelClientWriter.cs

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Output.Models;
+using AutoRest.CSharp.Output.Models.Shared;
+using AutoRest.CSharp.Output.Models.Types;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using AutoRest.CSharp.Output.Models.Requests;
+
+namespace AutoRest.CSharp.Generation.Writers
+{
+    internal class LowLevelClientWriter
+    {
+        public void WriteClient(CodeWriter writer, LowLevelRestClient client, BuildContext context)
+        {
+            var cs = client.Type;
+            // Client type should have public constructor with equivalent parameters not taking ClientOptions type as last argument
+            writer.Line ($"#pragma warning disable AZC0007\n");
+
+            using (writer.Namespace(cs.Namespace + ".Protocol"))
+            {
+                writer.WriteXmlDocumentationSummary(client.Description);
+                using (writer.Scope($"{client.Declaration.Accessibility} partial class {cs.Name}"))
+                {
+                    WriteClientFields(writer, client, context);
+                    WriteClientCtors(writer, client, context);
+
+                    foreach (var clientMethod in client.Methods)
+                    {
+                        WriteClientMethod(writer, clientMethod, true);
+                        WriteClientMethod(writer, clientMethod, false);
+                        WriteClientMethodRequest(writer, clientMethod);
+                    }
+                }
+            }
+        }
+
+        private void WriteClientMethodRequest(CodeWriter writer, RestClientMethod clientMethod)
+        {
+            RequestClientWriter.WriteRequestCreation(writer, clientMethod, lowLevel: true);
+        }
+
+        private void WriteClientMethod(CodeWriter writer, RestClientMethod clientMethod, bool async)
+        {
+            var parameters = clientMethod.Parameters;
+
+            CSharpType? bodyType = clientMethod.ReturnType;
+            var responseType = async ? new CSharpType(typeof(Task<Response>)) : new CSharpType(typeof(Response));
+
+            if (async)
+            {
+                writer.UseNamespace("System.Threading.Tasks");
```

Comment:
You shouldn't need these UseNamespace calls now that you have `async ? new CSharpType(typeof(Task<Response>)) : new CSharpType(typeof(Response));` instead of strings.